sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java

```diff
@@ -21,17 +22,26 @@ public SentryWebFilterWithThreadLocalAccessor(final @NotNull IHub hub) {
   public Mono<Void> filter(
       final @NotNull ServerWebExchange serverWebExchange,
       final @NotNull WebFilterChain webFilterChain) {
+    final @NotNull TransactionContainer transactionContainer = new TransactionContainer();
     return ReactorUtils.withSentryNewMainHubClone(
         webFilterChain
             .filter(serverWebExchange)
             .doFinally(
-                __ -> {
-                  doFinally(Sentry.getCurrentHub());
-                  Sentry.setCurrentHub(NoOpHub.getInstance());
-                })
+                __ ->
+                    doFinally(
+                        serverWebExchange,
+                        Sentry.getCurrentHub(),
```

Comment:
Does `Sentry.getCurrentHub()` work correctly in this filter method? `ReactorUtils.withSentryNewMainHubClone` does not set the current hub, so shouldn't we get the hub from the context?
Or is this somehow handled by reactor behind the scenes?

Reply:
`ThreadLocal` is set via `ThreadLocalAccessor` and propagated via reactor context.

Reply:
Just tested again with debugger, works as expected (i.e. uses the correct hub).